packages/babel-plugin-transform-es2015-parameters/src/params.js

```diff
@@ -54,53 +54,41 @@ export default function convertFunctionParams(path, loose) {
   };
 
   const body = [];
-
-  let firstOptionalIndex = null;
-
-  //
   const params = path.get("params");
 
-  if (loose) {
-    const body = [];
-    for (let i = 0; i < params.length; ++i) {
-      const param = params[i];
-      if (param.isAssignmentPattern()) {
-        const left = param.get("left");
-        const right = param.get("right");
-
-        const undefinedNode = scope.buildUndefinedNode();
-
-        if (left.isIdentifier()) {
-          body.push(
-            buildLooseDefaultParam({
-              ASSIGNMENT_IDENTIFIER: left.node,
-              DEFAULT_VALUE: right.node,
-              UNDEFINED: undefinedNode,
-            }),
-          );
-          param.replaceWith(left.node);
-        } else if (left.isObjectPattern() || left.isArrayPattern()) {
-          const paramName = scope.generateUidIdentifier();
-          body.push(
-            buildLooseDestructuredDefaultParam({
-              ASSIGNMENT_IDENTIFIER: left.node,
-              DEFAULT_VALUE: right.node,
-              PARAMETER_NAME: paramName,
-              UNDEFINED: undefinedNode,
-            }),
-          );
-          param.replaceWith(paramName);
-        }
-      }
-    }
-    path.get("body").unshiftContainer("body", body);
-    return;
-  }
+  let firstOptionalIndex = null;
 
   for (let i = 0; i < params.length; i++) {
     const param = params[i];
 
-    if (param.isAssignmentPattern()) {
+    if (param.isAssignmentPattern() && loose) {
```

Comment:
Worth merging this with the next block that also wants isAssignmentPattern/left/right?

Reply:
well, i though about merging those 2 conditions, just couldnt stand the level of indentation it would have created ;) ive also wanted to extract it to some function for the readability but those blocks need access to just too many variables available in the scope

if you prefer I merge them just let me know and I'll amend my PR

Reply:
Yeah, had the same reservations re: nesting lol, it's fine.